lib/utils.js

```diff
@@ -320,4 +320,87 @@ utils.isUnsupportedQuery = function isUnsupportedQuery(queryObj) {
         .some(key => constants.unsupportedQueries[key]);
 };
 
+/**
+ * Check if the replication configuration exists and contains an applicable rule
+ * @param {string} objKey - The key of the object
+ * @param {object} [config] - The bucket replication configuration
+ * @return {object|undefined} The replication rule if found, otherwise
+ * `undefined`
+ */
+utils.getReplicationRule = function getReplicationRule(objKey, config) {
+    if (config) {
+        return config.rules.find(rule => objKey.startsWith(rule.prefix));
+    }
+    return undefined;
+};
+
+/**
+ * Get the object metadata replication information to set
+ * @param {string} config - The bucket replication configuration
+ * @param {object} rule - The replication rule
+ * @param {array} content - The array of content types to be replicated
+ * @return {object} The object metadata replication information
+ */
+utils.getReplicationInfo = function getReplicationInfo(config, rule, content) {
+    const replicationInfo = {
+        status: 'PENDING',
+        content,
+        destination: config.destination,
+    };
+    // StorageClass is optional for rules.
+    if (rule.storageClass) {
+        Object.assign(replicationInfo, { storageClass: rule.storageClass });
+    }
+    return replicationInfo;
+};
+
+/**
+ * Check the bucket replication configuration and set replication content
+ * @param {object} bucketMD - The bucket metadata
+ * @param {string} objKey - The key of the object
+ * @param {object} objMD - The object metadata
+ * @param {array} content - The content of the object to be replicated
+ * @return {undefined}
+ */
+utils.handleReplicationInfo =
```

Comment:
Sorry to be picky (again) I find this name confusing, and this function should stay internal instead of being part of the `utils` object. Maybe just call it `setReplicationInfoHelper` since it's not fundamentally different than the two other setters and is meant to be only a helper function.

Otherwise all other changes are good to me.